### PR TITLE
Toon shader update

### PR DESCRIPTION
**Pull request by s1lentssh**
### PURPOSE
Updated toon shader

### EFFECT OF CHANGE
Added 5 colors mode and transparency logic

### TECHNICAL STEPS
Added new toon inputs from RPR SDK
Added runtime creation of transparent and blend nodes (require storing rpr context inside node)

### NOTES FOR REVIEWERS
Notes, if needed.

### CODE REVIEW

File: pxr/imaging/rprUsd/materialNodes/rpr/toonNode.cpp [hunk 1]
```diff
@@ -110,6 +148,28 @@ class RprUsd_RprToonNode : public RprUsd_MaterialNode {
             }
 
             return ProcessInput<GfVec3f>(id, value, m_toonClosureNode, RPR_MATERIAL_INPUT_NORMAL);
+        } else if (id == _tokens->colorsMode) {
+            if (value.IsHolding<int>()) {
+                int colorModeInt = value.UncheckedGet<int>();
+                return m_rampNode->SetInput(RPR_MATERIAL_INPUT_TOON_5_COLORS, colorModeInt) == RPR_SUCCESS;
+            }
+            TF_RUNTIME_ERROR("Input `%s` has invalid type: %s, expected - `Token`", id.GetText(), value.GetTypeName().c_str());
```
Comment (hshakula) [inline pxr/imaging/rprUsd/materialNodes/rpr/toonNode.cpp:156]: expected token?

**Review by hshakula — APPROVED** [review]
I suppose that GUI is going to be quite dense with these changes - approximately 15 parameters. We will need to find a way to hide unneeded/unused parameters. This would require a different way of defining material nodes than we do right now and is definitely out of the scope of this PR.

File: pxr/imaging/rprUsd/materialNodes/rpr/toonNode.cpp (hunk 1 above)
Comment (bsavery) [reply pxr/imaging/rprUsd/materialNodes/rpr/toonNode.cpp:156]: ?

Comment (s1lentssh) [reply pxr/imaging/rprUsd/materialNodes/rpr/toonNode.cpp]: Updated error messages

**Pull request merged**